java-components/cli/src/main/java/io/github/redhatappstudio/jvmbuild/cli/rebuilt/RebuiltDownloadCommand.java

```diff
@@ -0,0 +1,191 @@
+                            ") from image " + rebuilt.get().getSpec().getImage());
+                    downloadImage(rebuilt.get().getSpec());
+                } else {
+                    System.out.println("Unable to find " + gav + " in list of artifacts");
+                }
+            } else if (artifact != null) {
+                Optional<RebuiltArtifact> rebuilt = Optional.ofNullable(builds.get(artifact));
+                if (rebuilt.isPresent()) {
+                    System.out.println("About to download " + artifact + " (for " + rebuilt.get().getMetadata().getName() +
+                            ") from image " + rebuilt.get().getSpec().getImage());
+                    downloadImage(rebuilt.get().getSpec());
+                } else {
+                    System.out.println("Unable to find " + artifact + " in list of artifacts");
+                }
+            } else if (sBom != null) {
+                if (!sBom.exists()) {
+                    System.out.println("Unable to find sbom " + sBom);
+                } else {
+                    try {
+                        Parser parser = new JsonParser();
+                        parser.parse(sBom).getComponents().forEach(c -> {
+                            String gav = c.getGroup() + ":" + c.getName() + ":" + c.getVersion();
+                            Optional<RebuiltArtifact> rebuilt = builds.values().stream()
+                                    .filter(b -> b.getSpec().getGav().equals(gav)).findFirst();
+                            if (rebuilt.isPresent()) {
+                                try {
+                                    downloadImage(rebuilt.get().getSpec());
+                                } catch (IOException e) {
+                                    throw new RuntimeException(e);
+                                }
+                            } else {
+                                System.out.println("Unable to find " + gav + " in list of artifacts");
+                            }
+                        });
+                    } catch (ParseException e) {
+                        throw new RuntimeException(e);
+                    }
+                }
+            }
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    private void downloadImage(RebuiltArtifactSpec spec)
+            throws IOException {
+
+        try (DockerClient dockerClient = DockerClientBuilder.getInstance().build()) {
+
+            String image = spec.getImage();
+            int lastColon = image.lastIndexOf(':');
+            String repository = image.substring(0, lastColon);
+            String tag = image.substring(lastColon + 1);
+
+            System.out.println("Downloading image from " + repository + " with tag " + tag);
+
+            // Pull the image from the remote registry
+            dockerClient.pullImageCmd(repository).withAuthConfig(dockerClient.authConfig()).withTag(tag).start()
```

Comment:
I would use Jib for this rather than the docker client, as AFAIK this requires a running docker demon on the users computer. An example would be the code here: https://github.com/redhat-appstudio/jvm-build-service/blob/b4031912abe34ec4a89425d43351628489bc6d4d/java-components/cache/src/main/java/com/redhat/hacbs/artifactcache/services/client/ociregistry/OCIRegistryRepositoryClient.java#L134